tests/Unit/CheckoutTest.php

```diff
@@ -319,10 +319,62 @@ public function testPaymentLinksSuccess($jsonFile, $httpStatus)
         $this->assertStringContainsString('payByLink.shtml', $result['url']);
     }
 
-    public static function successPaymentLinksProvider()
+    public static function successPaymentsLinkProvider()
     {
         return array(
-            array('tests/Resources/Checkout/payment-links-success.json', 200),
+            array('tests/Resources/Checkout/payment-links-success.json', 200)
+        );
+    }
+
+    /**
+     * @param string $jsonFile
+     * @param int $httpStatus
+     *
+     * @dataProvider expiredPaymentsLinkProvider
+     */
+    public function testPaymentLinksExpired($jsonFile, $httpStatus)
+    {
+        $client = $this->createMockClient($jsonFile, $httpStatus);
+
+        $service = new Checkout($client);
+
+        $params = array(
+            'status' => "expired"
+        );
+
+        $result = $service->updatepaymentLinks('linkid', $params);
+
+        $this->assertEquals('expired', $result['status']);
+    }
+
+    public static function expiredPaymentsLinkProvider()
+    {
+        return array(
+            array('tests/Resources/Checkout/payment-links-expired.json', 200)
```

Comment:
When you only have one case (and it's unlikely we'll add more), there is no need to use a data provider. Just inline these values in your test case.   

Data providers are useful when you have a "table" of several cases.

Reply:
Ah duly noted! I'll try to change it first thing Monday morning.